include/openssl/service_indicator.h

```diff
@@ -50,7 +50,7 @@ enum FIPSStatus {
     int before = FIPS_service_indicator_before_call();              \
     func;                                                           \
     int after = FIPS_service_indicator_after_call();                \
-    if (before != after) {                                          \
+    if (before + 1 == after) {                                      \
```

Comment:
`FIPS_service_indicator_before_call` and `FIPS_service_indicator_after_call` both return an unsigned int. This was on purpose because we didn’t want the counter to overflow to a negative number if we used int. We also wanted to make sure we behaved correctly when the unsigned `int` counter wrapped around to `0`, so we decided not to do `before + 1 == after` and go ahead with `before != after` at the time.
If we’re still concerned about this edge case, I don’t think we should change the original behavior. But I do agree we should add better testing around the counter to make sure that we’re only incrementing by `1`.  We should also document the original intention… Sorry, I had almost forgotten why we did this at the time.